clang/lib/CIR/Dialect/IR/CIRDialect.cpp

```diff
@@ -2888,21 +2888,41 @@ LogicalResult cir::FuncOp::verify() {
 
   std::set<llvm::StringRef> labels;
   std::set<llvm::StringRef> gotos;
-
+  std::set<llvm::StringRef> blockAddresses;
```

Comment:
Can you try to use small set from LLVM ADT? I know the previous ones need to probably be fixed, but lets not make it worse.